.travis/install.sh

```diff
@@ -3,54 +3,80 @@
 set -e
 set -x
 
-if [[ "$(uname -s)" == 'Darwin' ]]; then
-    export HOMEBREW_NO_AUTO_UPDATE=1
-    export HOMEBREW_LOGS=~/brew-logs
-    export HOMEBREW_TEMP=~/brew-temp
-    mkdir $HOMEBREW_LOGS
-    mkdir $HOMEBREW_TEMP
-    brew update > /dev/null
-    brew cleanup > /dev/null  # do this here, so it won't automatically trigger in the middle of other stuff
-    brew outdated pkg-config || brew upgrade pkg-config
-    # do NOT update openssl 1.0.x, brew will also update a lot of dependent pkgs (and their dependencies) then!
-    #brew outdated openssl || brew upgrade openssl
-    export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig:$PKG_CONFIG_PATH"
-    brew install readline
-    export PKG_CONFIG_PATH="/usr/local/opt/readline/lib/pkgconfig:$PKG_CONFIG_PATH"
-    brew install zstd
+# Abort installation and don't run tests for pull requests if commit only changed the docs
+if [ "${TRAVIS_EVENT_TYPE}" = "pull_request" ]
+then
+    printf '%s\n' "Checking whether ${TRAVIS_COMMIT_RANGE} changed only docs"
+    git diff --name-only "${TRAVIS_COMMIT_RANGE}" | grep --quiet --invert-match --extended-regexp '(AUTHORS|README\.rst|^(docs)/)' || {
+        printf '%s\n' "Only docs were updated, stopping build process."
+        exit 0
+    }
+fi
```

Comment:
does this actually work when invoked from here?

Reply:
I don't know why it wouldn't? :thinking: What are you concerned about? I can run an isolated test in a fresh repo if you'd like to be sure.

Reply:
Doesn't that just terminate the install step, but would still do the test step?

Reply:
You mean that it doesn't immediately quit but just keeps running? No, I don't think so, as far as I know that's only true for `script`. But I'll test it to make sure. :)

Reply:
Oh, I see what you mean, now. `exit` from inside the script only terminates the script itself and Travis just goes on to continue with `run.sh`. That's what you meant, right? I think there are three ways to go about this:

- Move the test for docs-only change back into `.travis.yml`. But I don't think that's a good place to have code - that's why I moved it out of there in the first place.
- Elaborate way to relay the "termination" signal from the exit in `install.sh` back to `.travis.yml`. I think that's way too much effort for such a small issue. I'd say having the code inside `.travis.yml` would be preferable to this.
- Merge `install.sh` and `run.sh` into one `script.sh` which then fully terminates with the one `exit` call. `run.sh` effectively only contains one line of code anyway, so there is not much separation lost other than the fold in the Travis log.

What do you think? I'll see if I can come up with other ideas, too.

Reply:
guess moving it back to .travis.yml is the simplest and least-change solution.
otherwise we you also have to rename stuff, which is not nice for git history.

Reply:
:+1: